### PR TITLE
hwinfo: use more up-to-date pci.ids / usb.ids

### PR DESCRIPTION
**Pull request by Mic92**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (via cross-compiling)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Lassulus — APPROVED** [review]

File: pkgs/by-name/hw/hwinfo/package.nix [hunk 1]
```diff
@@ -56,13 +64,26 @@ stdenv.mkDerivation (finalAttrs: {
...
 
-  makeFlags = [
-    "LIBDIR=/lib"
-    "HWINFO_VERSION=${finalAttrs.version}"
-  ];
+  # The pci/usb ids in hwinfo are ancient. We can get a more up-to-date list simply by copying from systemd
+  preBuild = ''
+    # since we don't have .git, we cannot run this.
+    rm git2log
+    # taken from https://github.com/openSUSE/hwinfo/blob/c87f449f1d4882c71b0a1e6dc80638224a5baeed/src/ids/update_pci_usb
```
Comment (hsjobeki) [inline pkgs/by-name/hw/hwinfo/package.nix:74]: Does this comment belong to line 78 ?

**Review by hsjobeki — APPROVED** [review]

File: pkgs/by-name/hw/hwinfo/package.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/by-name/hw/hwinfo/package.nix]: moved.

**Comment by jopejoe1** [comment]
Maybe we should package [pcie.ids](https://github.com/pciutils/pciids) and [usb.ids](https://sourceforge.net/p/linux-usb/repo/HEAD/tree/trunk/htdocs/usb.ids) and use them?

**Comment by Mic92** [comment]
> Maybe we should package [pcie.ids](https://github.com/pciutils/pciids) and [usb.ids](https://sourceforge.net/p/linux-usb/repo/HEAD/tree/trunk/htdocs/usb.ids) and use them?

I prefer keeping things in sync with systemd so that udev agrees with us on the same names. We want to use this in https://github.com/numtide/nixos-facter

**Review by jopejoe1 — APPROVED** [review]
Makes sense to me

**Review by brianmcgee — APPROVED** [review]
lgtm

**Pull request merged**